### PR TITLE
Documentation for configuration options

### PR DESCRIPTION
**Pull request by c00kiemon5ter**
This merge request adds documentation for existing configuration options, but also new ones as needed by #437 (eIDAS support). 

The first commit is whitespace changes (and is what causes the noise in the diff). 
Looking at each commit separately will be easier.

### CODE REVIEW

File: doc/howto/config.rst [hunk 1]
```diff
@@ -408,6 +408,20 @@ Example::
     }
 
 
+want_response_signed
+""""""""""""""""""""
+
+Indicates that Authentication Responses to this SP should be signed.
```
Comment (jkakavas) [inline doc/howto/config.rst]: I would say "must be signed" . And/or we can clarify that if this is set to True, the SP will not consume any SAML Responses that are not signed.

File: doc/howto/config.rst [hunk 2]
```diff
@@ -437,6 +437,22 @@ Example::
     }
 
 
+allow_unsolicited
+"""""""""""""""""
+
+When set to true, IdP initiated SSO is allowed. This means that a user can
+start an authentication flow from the IdP, and once the user is authenticated
```
Comment (jkakavas) [inline doc/howto/config.rst]: This is a little misleading, the user normally selects an SP that they want to access before they authenticate to the IdP. 

I would rephrase that to : 
This means that this SP will consume unsolicited SAML Responses, i.e. SAML Responses for which it has not sent a respective SAML Authentication Request.

File: doc/howto/config.rst [hunk 3]
```diff
@@ -502,6 +502,45 @@ Example::
     }
 
 
+requested_attributes
+""""""""""""""""""""
+
```
Comment (jkakavas) [inline doc/howto/config.rst]: This is relevant only for the eIDAS SAML profile. 
A list of attributes that the SP requests from an eIDAS-Service (IdP). These attributes are included in the SAML Authentication Request that this SP sends as it is shown in the example below : 
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<saml2p:AuthnRequest
 Destination="https://eidas-service.eu/post"
 ID="_171ccc6b39b1e8f6e762c2e4ee4ded3a" IssueInstant="2015-04-30T19:25:14.273Z" Version="2.0"
 xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" xmlns:eidas="http://eidas.europa.eu/samlextensions">
.........
<saml2p:Extensions>
 <eidas:SPType>public</eidas:SPType>
 <eidas:RequestedAttributes>
 <eidas:RequestedAttribute Name="http://eidas.europa.eu/attributes/naturalperson/PersonIdentifier"
NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" isRequired="true"/>
 <eidas:RequestedAttribute Name="http://eidas.europa.eu/attributes/naturalperson/CurrentFamilyName"
NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" isRequired="true"/>
 <eidas:RequestedAttribute Name="http://eidas.europa.eu/attributes/naturalperson/CurrentGivenName"
NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" isRequired="true"/>
 <eidas:RequestedAttribute Name="http://eidas.europa.eu/attributes/naturalperson/DateOfBirth"
NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri" isRequired="true"/>
 </eidas:RequestedAttributes>
 </saml2p:Extensions>
.........
</saml2p:AuthnRequest>
```

Please note that this is different from the `required_attributes` and `optional_attributes` parameters that control the requested attributes in the metadata of an SP.

File: doc/howto/config.rst [hunk 4]
```diff
@@ -472,6 +472,36 @@ This kind of functionality is required for the eIDAS SAML profile
 > eIDAS-Connectors SHOULD NOT provide AssertionConsumerServiceURL.
 
 
+sp_type
+"""""""
+
```
Comment (jkakavas) [inline doc/howto/config.rst]: This parameter is only relevant for the eIDAS SAML profile.

File: doc/howto/config.rst [hunk 5]
```diff
@@ -472,6 +472,36 @@ This kind of functionality is required for the eIDAS SAML profile
...
+    "service": {
+        "sp": {
+            "sp_type": "private",
+        }
+    }
+
+
+sp_type_in_metadata
+"""""""""""""""""""
+
```
Comment (jkakavas) [inline doc/howto/config.rst]: This parameter is only relevant for the eIDAS SAML profile.

**Comment by c00kiemon5ter** [comment]
Updated according to the commends by @jkakavas

**Comment by jkakavas** [comment]
I would propose we merge #437 first so that we won't introduce documentation for options that are not yet there

File: doc/howto/config.rst [hunk 6]
```diff
@@ -408,6 +408,21 @@ Example::
     }
 
 
+want_response_signed
+""""""""""""""""""""
+
+Indicates that Authentication Responses to this SP must be signed. If set to
+True, the SP will not consume any SAML Responses that are not signed.
```
Comment (jkakavas) [inline doc/howto/config.rst]: Mental note to add 
> Default value is `True`

when/if #439 is approved and merged
